tests/test_mbrl.py

```diff
@@ -0,0 +1,206 @@
+from emote.models.model_env import ModelEnv
+from emote.sac import FeatureAgentProxy
+
+
+class FakeDataloader:
+    def __init__(self, data_group: str, num_obs: int, batch_size: int):
+        self.data_group = data_group
+        self.num_obs = num_obs
+        self.batch_size = batch_size
+
+    def __iter__(self):
+        while True:
+            batch = {
+                "observation": {"obs": torch.rand(self.batch_size, self.num_obs)},
+            }
+            yield {self.data_group: batch, "batch_size": self.batch_size}
+
+
+class MultiplierNN(nn.Module):
+    def __init__(self, value: float, device: torch.device):
+        super().__init__()
+        self.value = value
+        self.device = device
+
+    def forward(self, x: torch.Tensor):
+        return self.value * x
+
+    def sample(self, x, rng):
+        return self.forward(x)
+
+
+class RandomPolicy(nn.Module):
+    def __init__(self, action_dim: int):
+        super().__init__()
+        self.action_dim = action_dim
+
+    def forward(self, obs: torch.Tensor):
+        batch_size = obs.shape[0]
+        rand_actions = 2 * (torch.rand(batch_size, self.action_dim) - 0.5)
+        return rand_actions, 0
+
+
+def create_memory(
+    env: DictGymWrapper,
+    memory_size: int,
+    len_rollout: int,
+    batch_size: int,
+    data_group: str,
+    device: torch.device,
+):
+    """Creates memory and data_loader for the RL training
+
+    Arguments:
+        env (DictGymWrapper): the Gym-env wrapper
+        memory_size (int): the maximum length of memory
+        len_rollout (int): the rollout size for the NStepTable
+        batch_size (int): batch size
+        data_group (str): the data group for uploading the data
+        device (torch.device): the device to upload the data
+    Returns:
+        (tuple[TableMemoryProxy, MemoryLoader]): A proxy for the memory and a dataloader
+
+    """
+    table = DictObsTable(
+        spaces=env.dict_space,
+        use_terminal_column=False,
+        maxlen=memory_size,
+        device=device,
+    )
+    memory_proxy = TableMemoryProxy(table=table, use_terminal=False)
+    data_loader = MemoryLoader(
+        table=table,
+        rollout_count=batch_size // len_rollout,
+        rollout_length=len_rollout,
+        size_key="batch_size",
+        data_group=data_group,
+    )
+    return memory_proxy, data_loader
+
+
+def test_model_collector():
+    """The function tests unrolling a dynamic model and storing the rollouts in a replay buffer.
+    The fake dynamic model simply multiplies the inputs by a fixed (rand) number, i.e.,
+        next_obs = obs x rand_number,
+        rewards = actions x rand_number.
+
+    The test checks the following:
+        * the replay buffer contains a correct number of samples,
+        * stored samples are the ones generated by the fake model.
+
+    """
+    num_obs = 2
+    num_actions = 1
+    batch_size = 10
+    rollout_size = 5
+    rl_data_group = "rl_group"
```

Comment:
Are these constants or config? I'd consider moving them out of the function and capitalize them to denote them as "constants".

Reply:
num_obs, num_actions and rl_data_group are constants and are moved outside functions. However, batch_size, and rollout_size have different values for the two functions. So, they are kept inside the functions.